File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webapi/service/types/WriteContext.java
package org.opcfoundation.webapi.service.types;

import org.eclipse.milo.opcua.stack.core.types.structured.RequestHeader;
import org.eclipse.milo.opcua.stack.core.types.structured.WriteValue;
import org.springframework.lang.Nullable;
import org.springframework.web.context.request.NativeWebRequest;

import java.util.List;

public class WriteContext extends ServiceContext {
    final private List<WriteValue> nodesToWrite;

    public WriteContext(
            List<WriteValue> nodesToWrite,
            @Nullable String serverUri,
            @Nullable RequestHeader header,
            @Nullable NativeWebRequest webRequest)
    {
        super(serverUri,header,webRequest);
        this.nodesToWrite = nodesToWrite;
    }

    public List<WriteValue> getNodesToWrite() {
        return nodesToWrite;
    }
}

File: opcua-webservice-node/src/server/types/contexts/WriteContext.ts
import { WriteRequest, WriteValue } from "opcua-webapi";
import { ServiceContext } from "./ServiceContext";

export class WriteContext extends ServiceContext
{
    private _nodesToWrite: Array<WriteValue>;
    
    constructor(request: WriteRequest, serverUri?: string)
    {
        super(serverUri, request.RequestHeader);
        this._nodesToWrite = (request.NodesToWrite) ? request.NodesToWrite : [];
    }

    get nodesToWrite() : Array<WriteValue>
    {
        return this._nodesToWrite;
    }
}